test/onnx/expect/TestOperators.test_slice_dynamic.expect

```diff
@@ -167,7 +167,7 @@ graph {
         elem_type: 1
         shape {
           dim {
-            dim_value: 0
+            dim_param: "Gather16_dim_0"
```

Comment:
This is expected:
before constant folding, the graph is
```python
graph(%0 : Float(3, 4, strides=[4, 1], requires_grad=1, device=cpu)):
  %1 : Long(2, strides=[1], device=cpu) = onnx::Shape(%0)
  %2 : Long(device=cpu) = onnx::Constant[value={0}]()
  %3 : Long(device=cpu) = onnx::Gather[axis=0](%1, %2) # /home/david/dev/pytorch_rel/test/onnx/test_operators.py:504:0
  %4 : Long(2, strides=[1], device=cpu) = onnx::Shape(%0)
  %5 : Long(device=cpu) = onnx::Constant[value={1}]()
  %6 : Long(device=cpu) = onnx::Gather[axis=0](%4, %5) # /home/david/dev/pytorch_rel/test/onnx/test_operators.py:504:0
  %7 : Long(requires_grad=0, device=cpu) = onnx::Constant[value={3}]()
  %8 : Long(requires_grad=0, device=cpu) = onnx::Sub(%6, %7)
  %9 : Long(requires_grad=0, device=cpu) = onnx::Constant[value={0}]()
  %10 : Long(requires_grad=0, device=cpu) = onnx::Constant[value={9223372036854775807}]()
  %11 : Long(1, strides=[1], device=cpu) = onnx::Unsqueeze[axes=[0]](%3)
  %12 : Long(1, strides=[1], device=cpu) = onnx::Unsqueeze[axes=[0]](%10)
  %13 : Long(1, strides=[1], device=cpu) = onnx::Unsqueeze[axes=[0]](%9)
  %14 : Long(1, strides=[1], device=cpu) = onnx::Constant[value={1}]()
  %15 : Float(0, 4, strides=[4, 1], requires_grad=1, device=cpu) = onnx::Slice(%0, %11, %12, %13, %14) # /home/david/dev/pytorch_rel/test/onnx/test_operators.py:504:0
  %16 : Float(0, strides=[4], requires_grad=1, device=cpu) = onnx::Gather[axis=1](%15, %8) # /home/david/dev/pytorch_rel/test/onnx/test_operators.py:504:0
  return (%16)
```
After const folding and before the final round of shape inference, the graph changes to three inputs:
```python
graph(%0 : Float(3, 4, strides=[4, 1], requires_grad=1, device=cpu),
      %17 : Long(1, strides=[1], requires_grad=0, device=cpu),
      %18 : Long(1, strides=[1], requires_grad=0, device=cpu)):
  %1 : Long(2, strides=[1], device=cpu) = onnx::Shape(%0)
  %2 : Long(device=cpu) = onnx::Constant[value={0}]()
  %3 : Long(device=cpu) = onnx::Gather[axis=0](%1, %2) # /home/david/dev/pytorch_rel/test/onnx/test_operators.py:504:0
  %4 : Long(2, strides=[1], device=cpu) = onnx::Shape(%0)
  %5 : Long(device=cpu) = onnx::Constant[value={1}]()
  %6 : Long(device=cpu) = onnx::Gather[axis=0](%4, %5) # /home/david/dev/pytorch_rel/test/onnx/test_operators.py:504:0
  %7 : Long(requires_grad=0, device=cpu) = onnx::Constant[value={3}]()
  %8 : Long(requires_grad=0, device=cpu) = onnx::Sub(%6, %7)
  %11 : Long(1, strides=[1], device=cpu) = onnx::Unsqueeze[axes=[0]](%3)
  %14 : Long(1, strides=[1], device=cpu) = onnx::Constant[value={1}]()
  %15 : Float(0, 4, strides=[4, 1], requires_grad=1, device=cpu) = onnx::Slice(%0, %11, %17, %18, %14) # /home/david/dev/pytorch_rel/test/onnx/test_operators.py:504:0
  %16 : Float(0, strides=[4], requires_grad=1, device=cpu) = onnx::Gather[axis=1](%15, %8) # /home/david/dev/pytorch_rel/test/onnx/test_operators.py:504:0
  return (%16)
```
But shape inference logic cannot extract the constant value from inputs: %17 and %18 here.
Therefore, the onnx graph becomes:
```python
graph(%0 : Float(3, 4, strides=[4, 1], requires_grad=1, device=cpu),
      %17 : Long(1, strides=[1], requires_grad=0, device=cpu),
      %18 : Long(1, strides=[1], requires_grad=0, device=cpu)):
  %1 : Long(2, strides=[1], device=cpu) = onnx::Shape(%0)
  %2 : Long(device=cpu) = onnx::Constant[value={0}]()
  %3 : Long(device=cpu) = onnx::Gather[axis=0](%1, %2) # /home/david/dev/pytorch_rel/test/onnx/test_operators.py:504:0
  %4 : Long(2, strides=[1], device=cpu) = onnx::Shape(%0)
  %5 : Long(device=cpu) = onnx::Constant[value={1}]()
  %6 : Long(device=cpu) = onnx::Gather[axis=0](%4, %5) # /home/david/dev/pytorch_rel/test/onnx/test_operators.py:504:0
  %7 : Long(requires_grad=0, device=cpu) = onnx::Constant[value={3}]()
  %8 : Long(requires_grad=0, device=cpu) = onnx::Sub(%6, %7)
  %11 : Long(1, strides=[1], device=cpu) = onnx::Unsqueeze[axes=[0]](%3)
  %14 : Long(1, strides=[1], device=cpu) = onnx::Constant[value={1}]()
  %15 : Float(*, *, strides=[4, 1], requires_grad=1, device=cpu) = onnx::Slice(%0, %11, %17, %18, %14) # /home/david/dev/pytorch_rel/test/onnx/test_operators.py:504:0
  %16 : Float(*, strides=[4], requires_grad=1, device=cpu) = onnx::Gather[axis=1](%15, %8) # /home/david/dev/pytorch_rel/test/onnx/test_operators.py:504:0
  return (%16)
```
Thinking to remove symbolic shape inference in the final round of shape inference.

Reply:
Added SetValue logic as the first step of shape inference block, this issue is resolved.